src/fcntl.rs

```diff
@@ -210,6 +210,55 @@ pub fn renameat<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
+#[cfg(all(
+    target_os = "linux",
+    not(target_env = "musl"),
+    any(
+        target_arch = "x86_64",
```

Comment:
The CI failed on numerous targets, as per: https://github.com/nix-rust/nix/pull/1458/checks?check_run_id=2950043941

I'm not sure if there was some other problem causing that at the time, and so I'll try and undo those restrictions.  I've just discovered the `cross` utility so I can run the tests locally instead of pushing up and waiting for the CI tasks.  Give me a little while.

Reply:
So, I ran with `all(target_os = "linux", target_env = "gnu")`, and targeted MIPS64, and this is the result.  I have no expertise here.  I don't know if the problem is with the libc crate or whatever else!

```
cross test --target mips64-unknown-linux-gnuabi64
   Compiling libc v0.2.97
   Compiling proc-macro2 v1.0.27
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.73
   Compiling autocfg v1.0.1
   Compiling bitflags v1.2.1
   Compiling libc v0.2.97 (https://github.com/rust-lang/libc?rev=72da7a6a464a5a2a17993b94d94d2ab99bb87db6#72da7a6a)
   Compiling semver v1.0.3
   Compiling memoffset v0.6.4
   Compiling quote v1.0.9
   Compiling getrandom v0.2.3
   Compiling errno v0.2.7
   Compiling rand_core v0.6.3
   Compiling rand_chacha v0.3.1
   Compiling rand v0.8.4
   Compiling nix v0.21.0 (/project)
   Compiling tempfile v3.2.0
   Compiling thiserror-impl v1.0.25
   Compiling thiserror v1.0.25
   Compiling caps v0.5.2
error: linking with `mips64-linux-gnuabi64-gcc` failed: exit status: 1
  |
  = note: "mips64-linux-gnuabi64-gcc" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.107qzeinzma9qkkx.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.10ufato51vgf1mih.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.12m217bo86ad2rkc.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.12xswyq6a7re2wn7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.16bfp76jmzn2jafv.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.16cfaj94tkdl1hrw.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.16f3gp9924cddqt0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.16wtaxp2uqq04ygw.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.18dwho9kd77aop5u.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.18qk0yugj4at5ruc.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.19ymatn6y37zsn1z.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1afh56gw6cswbuhi.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1azssnm1yids34lb.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1ci7cj5d1rvb8crl.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1cwcacunsx6m1om1.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1deb0sc8autqpuh3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1dhymmwh6mtrg6ux.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1ekup1naeh8kzu6g.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1eokme0c26pydjfg.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1g518ehn083z1t0e.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1g57ofar520aymya.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1hbvmmgf03ftfviv.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1hii2lhb4aidp6dy.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1hm5rvehaes90yz0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1hme8h4yub1l35rr.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1i5gc33zoqt8lvuf.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1ieiy5eifvjwtlwp.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1j92yad28s4sqxyz.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1k7rw7cetb73tcai.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1koirrmgkd3xozfl.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1l96gsz5xu0lbkf7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1lx0oponiptg4w0f.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1mmz9irvmc078t0e.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1mydoq5q1jpy13pv.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1n5ie4dd8hp09juh.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1nczl4bp0r3ye736.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1netbh09y9ws356a.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1ovlq6y1ofhkb89k.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1oyrnjozuy04bby3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1q3tqn96dypcciiz.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1qfi0ghnglbn010y.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1ql2mdur17v8gk8a.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1rj93vafkvb4yls3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1somlc4ns201amh7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1tln9q1gj1c9ifte.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1uey5fgu2qk49t9v.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1um85im7881non50.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1umc06wr21ey8pqj.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1uq7hdnayrq3g9n9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1xx4lmtyustx694w.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1yh3f2n25c0qw6ah.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1ym3ryz0r376jhsl.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1z7lukvr21gprgr7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1z8hemfnskkl798f.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1z9mppmdcyve64b0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1zf8izvy70mrx71u.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1zxs8vgrjevxpjfa.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2052ghw4a9thhzsq.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.208hof62fd3z2fkn.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.20e5mjsgr9to5kjz.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.212u4f52h0l2eow2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.21ozzv96rf3hccps.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.239qw24m6fz1ukj0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.25z2spnvhhg59ftq.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.262aix5x7jo4koxt.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.26vlk4ro7kax1vn6.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.276glaujkq34o5db.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.28nvhs0j1m34sty3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.29nxc1nfb3e58bqe.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2a4wg3h1fvvr3dyh.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2af5ogfynrk3yoad.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2dcrvorjsx94vzxl.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2dj1hty81vocodbp.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2djhqdd33cgxp8nn.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2gnp5k19vruqsmfo.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2htkisjib9znv98p.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2iudzwf0qv2w9p70.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2iultkc2sof0ahjz.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2k72wszp49oo3i0l.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2kbvj26czwvf1l94.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2nc06tv3s0f0mnke.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2olcpmijlkmy9wlf.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2oq9iom4chtb1lbf.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2r6utsi1r3vgy6bm.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2snfx120wkrliejv.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2t0y4hm4kb75om0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2t46r2uycxkb1jz.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2t7edbl70266iwe9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2to41580d6v3untm.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2v2hepflfswcdrt9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2vmtum3qemhqmort.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2x6ip18g5cmqk20v.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2xpyt0ewmdc635rj.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2ybdquauztm9rmt2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2z5p0b5t02m3l7wg.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2zqjkaor6f7uznjx.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.31hwycyapih2hin8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.324jpf3pcm91qdvt.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.32gijnxavbjold2j.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.32vnkxf62mg3ckkg.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.35ahwrfempc3w13r.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.35xaqfh0j8ityx6d.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3681wqej8aai58xu.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.369s240qmabhzox0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.37q7x4hlimwcj7vt.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.37x4tso8j25i6yh7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.38o5k7yb3so57mgg.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.38y5ubpblik1t0g3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.39ddi0o0hb04a8nr.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.39k6rtlrabrv3ntm.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.39wxr17ey2i46e7r.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3a1976zsgv41lome.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3afepnxr8xsvc9j6.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3aujo8tqx1mfib6r.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3awnvkwf1hpdm6v0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3c4xxarxo73a1xo4.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3c9fklewuzxxpkcb.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3gazqfgm3whckw1c.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3gs10lhyz173kwme.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3hh4559mupami9vs.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3hkyoqrs7xrzpiqq.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3ia9tl9frcxixqzu.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3inggbjrmq0ryc73.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3kl718m0649cocpb.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3l9ji0wtstn6y10g.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3mn9a62pjfj80hsa.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3n1wiji84i3d9flm.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3rdqkgofz994qexd.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3rqmpa4cfmd20qjf.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3scrzhs1l7rbvfru.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3ug1cd2rszbhzihy.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3vd141k1gg17mqch.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3vjcuuynfuj35wr2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3x8a495f1rrf6hl0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3xpjlwr9be11bjs9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3yie2vemi1b0vaa4.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3z8zxl82az2bwiik.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3ze9jhj57mll684s.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3zhx430z2z0fjtet.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.3zkcor5lrslguc34.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4054ajf67eg42jer.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4159c4b1awevnfo8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.41fxlr5r59ayoksk.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.41nffn87enbb84qb.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.41yf4c6xxgz3rvmw.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.422z6srmbphcdxx7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.42iy6p8451yyw4ug.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.45phhew5uplf5t2i.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.471r5n4dn96miwbu.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.47l500yw9b9snlli.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.480mg1kys3bneuzr.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4812cd59bvkkt6i6.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.48u9dt30kzcblc16.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.49em2bzmrsutgx1m.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4ahp0yrd3l8ds7wa.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4akoq93qo7tyb23q.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4anik8tof8itvs45.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4anq1eaw3p3ehm8c.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4ay3bkf8zxiop0te.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4bh3sepuzf2e6ser.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4bs4zwms2r6809y2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4bzbysp582rljrvn.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4dhncw0njm17nom7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4e752weocmyn7gn3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4eqdgo7w2t7wxd5j.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4f4jlcgqln3n3rfb.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4fpqm3pnxkzu5ru0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4g3vigonq8rjz0s2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4gh3im8847zjxkzo.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4hmf64zou730e2f0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4hxpjv4l3afuen7g.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4i5q9puzpwyayq71.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4imcw7zp4omsdo7r.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4iv9k4n85ddyojs2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4jeo5r164b937ils.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4kkcnt5ohdt8ekqd.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4kmbkn0xzqpib1tv.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4kz3hsea0olka84n.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4mtuxdf9vxhas9j7.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4nqd6foz9gz1043l.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4q7jcatm3ct9kav8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4radnzevjfmah1j9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4s2a7k3tc5ru9217.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4tdtff8h5dk19q1d.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4ueavusbs5oo6vhc.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4vrziw68d2pwgx5n.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4w9u375iv022x2gp.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4x50o125kjam3umu.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.4zzj9mhfsfvzpvw.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.50xzjivbmfnx5hd8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.50zgcseuw7pkc8mp.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.518po792tr07zxrw.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.521v4rnh1zox4y49.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.52v4r9gj17g3r7sl.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.53d0ibnwihes9mnq.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.53touc89n966axs2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.53uw9u3x26j94vy9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.55hs7s4gb0lxxbie.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.55lgggnuaol4ro7k.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.55n2p6limt5zb91q.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.56y53rllknvta9sl.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.57cote3e9c9io06g.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.57gf48p5nz3hyvl3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.585lxw92m0qy7ykn.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.58i128xvth7dt140.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.59825h8l5dvo4rmt.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.59faqek738y6f2ir.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5a7dwpxh02lvfqob.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5ae30nyursoo1z04.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5bduwna69fppv1y9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5beyqehxo5b3np5w.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5cmwhaxfrti27pd8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5dmbe2ivghvybnje.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5ds7jdwg8ka80m3z.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5ds7k8l6yge8ri59.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5dy6nrvaovs0txw6.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5e00f01zjjactg8q.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5e9u7edojf1d1f49.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5gaxz6a71mzjbp5l.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5gdgy7mzpzqmxjsk.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.5ggztr3ahapklrz1.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.657e616hpj158v3.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.7d0nlznyrlifh0v.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.7jyzvijaxnfyebn.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.8wzmgmc5fhvv93k.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.a0zk9n81bho80b9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.bgu6zlnh0mya9y9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.c8gb0xviile4ip0.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.clpkpk1rxnext7o.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.cnksus2i3nbrwah.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.d5okxny7cxcbqs8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.dveo4zdnay98phi.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.e72jdyw9ja2u9l9.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.eum550ibo7rg1ar.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.fhmix1wc2kbcuaa.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.fof0evhmfksbti8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.iiv9lg3hgvitfqy.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.jofnb0l7zbew3m.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.jp001uobrjhcfym.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.k5fuok9uhpfpmdh.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.kn3ac7rqxu6lyge.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.kyg76au28e5darq.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.l7w77ts9kfsfud.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.mnnpizw3sa59ni2.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.nmyvt9r0lw0gtkd.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.nxgsbhc63o7qhpp.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.nzo1wfskzu4zyk4.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.o1uc4xwv2qzdxs4.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.o7mvrmmssqi12qa.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.qfqqnyp808i6dvt.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.s39o27bm5yns5mg.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.snzhbeboiji9j19.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.tetdp215q3bdhs8.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.uk9h870aqnvn0ht.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.w34b8qd25a7tls1.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.zvc0mjgecln70ur.rcgu.o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.2h17trl0x2ius0s3.rcgu.o" "-Wl,--as-needed" "-L" "/target/mips64-unknown-linux-gnuabi64/debug/deps" "-L" "/target/debug/deps" "-L" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib" "-Wl,-Bstatic" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libcaps-7abee3df1c2531db.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/liberrno-ebd171d2a2a95e5d.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libthiserror-fc1cfab8787cc407.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libsemver-6660e48978b9c6e4.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libtest-3d51270a9e7f20a6.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libterm-f0534737d2624d30.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libgetopts-a350a38d6aaac23a.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libunicode_width-3ac41a23a2ed3f8b.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/librustc_std_workspace_std-69aca80aa7d5a46a.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libtempfile-182060886664078d.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/librand-b9028719376d99ba.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/librand_chacha-ab2b00778a0b7858.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libppv_lite86-91ff1554d9de749d.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/librand_core-e4eb3c0718d1f0f2.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libgetrandom-36ff4353ade419ae.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/liblibc-dcf0a81487d308f3.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libremove_dir_all-61d51878b6183bf2.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/liblazy_static-77f1048f0cf35aa3.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libnix-eec33d957463c3ca.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libbitflags-05f436ca68bdba93.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libmemoffset-6bc4eef0778036c4.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/liblibc-4f066077a72c8d7a.rlib" "/target/mips64-unknown-linux-gnuabi64/debug/deps/libcfg_if-fa9e552a9411e41a.rlib" "-Wl,--start-group" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libstd-d0efac7dd9adee97.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libpanic_unwind-c3d5935c549eb5ae.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libminiz_oxide-789f7c3338db59e0.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libadler-439005e25becd618.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libobject-3073f8a682713d2a.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libaddr2line-a05cffc80fd83586.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libgimli-e2d0269045b7904f.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libstd_detect-90ebf4a4b3120cd2.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/librustc_demangle-9eee76f0e50a87a4.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libhashbrown-0d81838d24879545.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/librustc_std_workspace_alloc-df04325256fcf9dd.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libunwind-e795af396ae34e31.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libcfg_if-f68928f04f1cd331.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/liblibc-5da8f35e5e96d4f5.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/liballoc-a4ebc47902a436af.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/librustc_std_workspace_core-af113e15d5b8aeda.rlib" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libcore-afd374598d7c7b14.rlib" "-Wl,--end-group" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib/libcompiler_builtins-9309922c7e5b8f86.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/rust/lib/rustlib/mips64-unknown-linux-gnuabi64/lib" "-o" "/target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs"
  = note: /target/mips64-unknown-linux-gnuabi64/debug/deps/test-c88adc2139c3600d.1yh3f2n25c0qw6ah.rcgu.o: In function `nix::fcntl::renameat2::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h429a86d48485915d':
          /project/src/fcntl.rs:238: undefined reference to `renameat2'
          /project/src/fcntl.rs:238: undefined reference to `renameat2'
          collect2: error: ld returned 1 exit status


error: aborting due to previous error

error: could not compile `nix`
```

Reply:
Interestingly, `cross` fails in basically the same way when targeting `x86_64-unknown-linux-gnu`

```  = note: /target/debug/deps/test-28090f7b17e1c885.1iqpr2alr2hxgxm4.rcgu.o: In function `nix::fcntl::renameat2::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h84a52447531af3c2':
          /project/src/fcntl.rs:238: undefined reference to `renameat2'
          collect2: error: ld returned 1 exit status```

Reply:
It looks to me like the CI runs in an entirely different way for the targets that were succeeding in the CI run I linked to.  The arches that were failing look like they could be running with `cross`, like I've just been trying, but linux x86_64, which is failing in `cross` for me, looks like it's doing something entirely different in the CI, see here: https://github.com/nix-rust/nix/pull/1458/checks?check_run_id=2950043979

Reply:
Sorry to bombard you, but when I run the tests simply with `cargo test`, they get to this test and hang perpetually: ```test sys::test_socket::test_af_alg_aead has been running for over 60 seconds```

Reply:
Yes, x86_64 runs the usual way in CI; simply "cargo test".  But other arches need QEMU, so they run with cross.  I'm guessing that cross's version of glibc is too old.  `renameat2` only appeared in glibc 2.28.  Have you figured out what version of glibc is included in cross?

Reply:
Yes, that's it.  rust-embedded/cross is using an older version of glibc.  And I don't think they're going to update, because they're focused on maximum portability I doubt they will upgrade.  What happens if you leave `renameat2` enabled for all architectures, but disable the tests on cross-tested arches?
https://github.com/rust-embedded/cross

Reply:
Ah great, so `cross` seems happy enough to compile with the syscall enabled, and the tests disabled.  I've just pushed up those changes.  We shall see in a few minutes!